frontend_build/src/extract_$trs.js

```diff
@@ -0,0 +1,72 @@
+var esprima = require('esprima');
+var logging = require('./logging');
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var path = require('path');
+
+function extract$trs(messageDir, messagesName) {
+  this.messageDir = messageDir;
+  this.messagesName = messagesName;
+}
+
+extract$trs.prototype.apply = function(compiler) {
+
+  var self = this;
+
+  compiler.plugin("emit", function(compilation, callback) {
+    var messageExport = {};
+
+    compilation.chunks.forEach(function(chunk) {
+      // Explore each module within the chunk (built inputs):
+      chunk.modules.forEach(function(module) {
+        if (module.resource && module.resource.indexOf('.vue') === module.resource.length - 4) {
+          // Explore each source file path that was included into the module:
+          var messageNameSpace;
+          var messages = {};
+          var ast = esprima.parse(module._source.source());
+          ast.body.forEach(function (node) {
```

Comment:
Can you explain at a high level, what are these AST-related statements doing? What's the high level goal?